packages/nextjs/src/app/createTenantResolverForClientProvider.ts

```diff
@@ -0,0 +1,32 @@
+import { CustomLoginOptionsType, CustomLoginParamKeyType, CustomLoginSubDomainType } from '../types';
+
+const isCustomLoginStrategyParamKey = (
+  customLoginOptions: CustomLoginOptionsType
+): customLoginOptions is CustomLoginParamKeyType => customLoginOptions.strategy === 'paramKey';
+
+const isCustomLoginStrategySubDomain = (
+  customLoginOptions: CustomLoginOptionsType
+): customLoginOptions is CustomLoginSubDomainType => customLoginOptions.strategy === 'subDomain';
+
+const emptyTenantResponse = { tenant: null };
+export const createTenantResolverForClientProvider = (customLoginOptions?: CustomLoginOptionsType) => {
+  if (!customLoginOptions) {
+    return undefined;
+  }
+  return () => {
+    try {
+      if (isCustomLoginStrategySubDomain(customLoginOptions)) {
+        const { subDomainIndex } = customLoginOptions;
+        return { tenant: window.location.hostname.split('.')[subDomainIndex] };
+      } else if (isCustomLoginStrategyParamKey(customLoginOptions)) {
+        const { paramKey } = customLoginOptions;
+        const params = new URLSearchParams(window.location.search);
+        const tenant = params.get(paramKey);
+        return { tenant };
+      }
+      return emptyTenantResponse;
```

Comment:
Empty tenant resposne must be empty object. 
String -> we have value
null -> invlaid value
undefined / empty object -> not configured